DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/README.md

```diff
@@ -0,0 +1,252 @@
+  - Shift values greater than the width of the data types are treated as a shift equal to the width of the data type.
+  - The shift operation can be done more efficiently by specifying the amount to shift with the smallest possible `ac_int`.
+
+### Bit Select Operator
+
+The bit select operator `[]` allows reading and modifying bits in an `ac_int`.
+
+*Note:* An `ac_int` must be initialized before being access by bit select operator `[]`, otherwise, it is undefined behavior and will give you unexpected results.
+
+For full details, see the Algorithmic C (AC) Datatypes documentation available at https://hlslibs.org/.
+
+### Bit Slice Operations
+
+The slice read operation `slc` and the slice write operation `set_slc` allows reading and modifying a slice in an `ac_int`.
+
+Slice read is provided with the template function `slc<int W>(int lsb)`. The two arguments are defined as:
+- `W` is the bit length of slice. It is constrained to be static so that the length of the slice is known at compile time.
+- `lsb` is the index of the LSB of the slice being read.
+
+Slice write is provided with the function `set_slc(int lsb, const ac_int<W, S> &slc)`. The two arguments are defined as:
+- `lsb` is the index of the LSB of the slice being written.
+- `slc` is an `ac_int` slice. The bit length of slice is inferred from the width `W` of `slc`.
+
+*Note:* An `ac_int` must be initialized before being access by bit slice operations `slc` and `set_slc`, otherwise, it is undefined behavior and will give you unexpected results.
+
+For full details, see the Algorithmic C (AC) Datatypes documentation available at https://hlslibs.org/.
+
+### Understanding the Tutorial Design
+
+This tutorial consists of five kernels as following:
+
+Kernel `BasicOpsInt` contains native `int` type addition, multiplication, and division operations, while kernel `BasicOpsAcInt` contains `ac_int` type addition, multiplication, and division operations. By comparing these two kernels, you will find reduced width `ac_int` generates more efficient hardware than native `int`.
+
+Kernel `ShiftOp` contains an `ac_int` left shifter and the data type of the shift amount is a large width signed `ac_int`. On contrast, kernel `EfficientShiftOp` also contains an `ac_int` left shifter, but the data type of the shift amount is a reduced width unsigned `ac_int`. By comparing these two kernels, you will find shift operations of `ac_int` can generate more efficient hardware if the amount to shift by is stored in a minimally sized unsigned `ac_int`.
+
+Kernel `BitOps` demonstrates bit operations with bit select operator `[]` and bit slice operations `slc` and `set_slc`.
```

Comment:
Do you prefer to remove the demonstration of bit operations for now until the reporting is fixed?

Reply:
I am not sure. Depends on what we plan to do about the case. If we don't plan to fix the reporting issue, maybe this would be a good place explain it. @ajaykumarkannan ?